src/blog/_posts/zonal-isolation-v1.md

```diff
@@ -0,0 +1,125 @@
+
+## What is Zonal Isolation for Cadence Workflows?
+At high-level, Zonal Isolation for Cadence Workflows can be thought in 2 levels:
+
+1. **Task-level isolation:** All decision tasks and activity tasks of a workflow are only processed by workers from the zone where the workflow was started
+2. **Infrastructure-level isolation:** Within a regional Cadence cluster, workflows are handled by server instances in the same zone where they were started, and the corresponding data is stored in that zone as well.
+
+Infrastructure-level isolation is quite challenging to implement as it requires significant changes to the core design of the Cadence server. Due to the complexity involved, support for this feature is not planned for the foreseeable future.
+
+As a result, the focus remains on achieving task-level zonal isolation outside the Cadence server, which offers a more practical and immediate way to improve system resilience. It provides the capability of ensuring that an unhealthy zone (i.e. bad deployment of workers) only affect a subset of workflows (started from a certain zone) rather than every workflow in a Cadence domain.
+
+
+## How Zonal Isolation Works in Cadence?
+
+### Architecture
+Here is what the architecture of a zonally isolated Cadence-based system looks like:
+![zonal isolation overview](/img/zone-isolation.png)
+*<p style="text-align:center;">Fig: Workflows started in one zone are only dispatched to workers from the same zone. Colors to emphasize pinning.</p>*
+### Implementation
+
+#### Determine the zone of a workflow and workers
+
+To ensure that tasks are dispatched to workers in the same zone as the workflows, we must identify the origin zone of both. The zone of a workflow is determined by the origin zone of the StartWorkflowExecution request, while the zone of workers is determined by the origin zone of the PollForDecisionTask and PollForActivityTask requests. There are three possible ways to determine the origin zone for these requests:
+
+- **Uber's Approach:** Let Cadence SDK set the origin zone in the headers of the requests before sending the request to Cadence.
+- **Preferred Approach:** Get the origin zone of the requests from headers set by network infrastructure.
+- Determine the origin zone of the requests from the zone of the cadence-frontend instance receiving the request, if the network layer has already achieved zonal isolation.
+
+The 2nd approach is the ideal one, but Uber's network infrastructure doesn't provide such headers and the network layer is not ready for zonal isolation. As a result, we adopt the 1st approach. At Uber, we have internal libraries in Go and Java acting as wrappers around Cadence SDK injecting necessary configurations. These libraries have been updated to include the origin zone in the request headers using a header called isolation-group.
```

Comment:
for those folks using OS cadence, can you explain what exact headers in the request they need to set?